packages/commons/src/risk-analysis/riskAnalysisValidation.ts

```diff
@@ -0,0 +1,326 @@
+import { TenantKind } from "pagopa-interop-models";
+import { P, match } from "ts-pattern";
+import {
+  RiskAnalysisFormToValidate,
+  RiskAnalysisValidatedForm,
+  RiskAnalysisValidatedSingleOrMultiAnswer,
+  RiskAnalysisValidationInvalid,
+  RiskAnalysisValidationResult,
+  ValidationRule,
+  ValidationRuleDependency,
+} from "./models.js";
+import {
+  RiskAnalysisValidationIssue,
+  dependencyNotFoundError,
+  missingExpectedFieldError,
+  noRulesVersionFoundError,
+  unexpectedDependencyValueError,
+  unexpectedFieldError,
+  unexpectedFieldFormatError,
+  unexpectedFieldValue,
+  unexpectedRulesVersionError,
+} from "./riskAnalysisErrors.js";
+import {
+  FormQuestionRules,
+  RiskAnalysisFormRules,
+  dataType,
+} from "./rules/models.js";
+import { riskAnalysisFormRules } from "./rules/riskAnalysisFormRules.js";
+
+export function validateRiskAnalysis(
+  riskAnalysisForm: RiskAnalysisFormToValidate,
+  schemaOnlyValidation: boolean,
+  tenantKind: TenantKind
+): RiskAnalysisValidationResult<RiskAnalysisValidatedForm> {
+  const latestVersionFormRules = getLatestVersionFormRules(tenantKind);
+
+  if (latestVersionFormRules === undefined) {
+    return invalidResult([noRulesVersionFoundError(tenantKind)]);
+  }
+
+  if (latestVersionFormRules.version !== riskAnalysisForm.version) {
+    return invalidResult([
+      unexpectedRulesVersionError(riskAnalysisForm.version),
+    ]);
+  }
```

Comment:
With assertions that assert something or throw, this pattern makes sense - but in these two specific cases it could make the code less readable because I cannot just throw in the separate method - I need to return the `invalidResult` here. If it makes sense to you, I would leave it as it is :)